chapter2.adoc

```diff
@@ -29,7 +31,12 @@ the RDSM (except for when explicitly allowed by the RDSM). Privilege levels may
 affect changes in the MTT under purview of the Supervisor Domain Security
 Manager (SDSM) either through an SBI interface into M-mode (or may have the
 ability to edit MTT structures by virtue of how the MTT structure in memory is
-accessible to lower privilege levels).
+accessible to lower privilege levels). Other memory access control mechanisms
+may also be used to enforce supervisor domain isolation for use cases that do
+not require the full flexibility of MTT. For example, (e)PMP configurations
+controlled by M-mode during domain context switching. In this case the MTT-mode
+`Other` may be used. When MTT mode is `Other`, MTT is not used, but the SDID
+CSR defined by `Smsdid` is used as defined.
```

Comment:
It is better to define that there is no option to disable memory protection provided by MTT but the MTT enforcement can be effectively disabled by setting `mode` to `Bare`. When MTT is in `Bare`, mode it does not enforce any memory protection. With this definition we do not need a `Other` encoding for MTT. It also does not require this text which would imply that e(PMP) are mutually exclusive with MTT whereas e(PMP) are not bypassed even with MTT is not `Bare`.

Reply:
Yes, this was my understanding following other tweaks made to the spec.

Basically "MTT mode" turns MTT on or off. All we need to say is that when MTT mode is "bare" (no MTT enforcement), other architectural memory isolation features can still be used to enforce physical memory isolation. 

And all the other parts of smmtt can still be used as well - SDID, interrupt extensions, debug extensions, etc. Only difference is MTT is disabled/not supported.

I don't know if IOMTT makes sense without MTT...

Reply:
The `mttp` CSR must always exists. The `mode` field of `mttp` is WARL so it is legal to hard wire the `mode`to `Bare`.  There are two functions provided by IOMTT:
- Supervisor domain classifier - classify a DMA request to a SDID (and MTT)
- MTT check
The supervisor domain classification is used to identify the IOMMU that should govern the DMA requests. So similar will likely apply that the MTT mode in IOMTT may be hard wired to `Bare` but the classifier to classify DMA requests to supervisor domains would exist.

Reply:
ack - if some other mechanism is defined in the future, such an "Other" option can be added at that point - so I will delete the change patch for the mttp.MODE CSR field and update the text above accordingly to specify use of 'Bare' mode.